docs/topics/native/native-app-with-c-and-libcurl.md

```diff
@@ -0,0 +1,260 @@
+        else throw new FileNotFoundException("Host OS is not supported in Kotlin/Native.")
+
+    nativeTarget.with {
+        binaries {
+            executable {
+                entryPoint = 'main'
+            }
+        }
+    }
+}
+```
+
+</tabs>
+
+* Targets are defined using `macOSX64`, `linuxX64`, and `mingwX64` for macOS, Linux, and Windows. For a complete
+  list of supported platforms, see the [Kotlin Native overview](native-overview.md#target-platforms).
+* The entry itself defines a series of properties to indicate how the binary is generated and the entry
+  point of the applications. These can be left as default values.
+* The C interoperability is configured as an additional step of the build. By default, all the symbols from C are
+  imported to the `interop` package; you may want to import the whole package in `.kt` files. Learn more about ways on
+  [how to configure](mpp-discover-project.md#multiplatform-plugin) it.
+
+## Create a definition file
+
+When writing native applications, you often need access to certain functionality that is not included in the [Kotlin
+standard library](https://kotlinlang.org/api/latest/jvm/stdlib/), such as making HTTP requests, reading and writing from
+disk, and so on.
+
+Kotlin/Native helps consume standard C libraries, opening up an entire ecosystem of functionality that exists
+for pretty much anything you may need. Kotlin/Native is already shipped with a set of prebuilt [platform libraries](native-platform-libs.md),
+which provide some additional common functionality to the standard library.
+
+An ideal scenario for interop is to call C functions as if you are calling Kotlin functions, following the same
+signature and conventions. This is when the `cinterop` tool comes in handy. It takes a C library and generates the
+corresponding Kotlin bindings, so that the library can be used as if it were a Kotlin code.
+
+To generate these bindings, let's create a library definition `.def` file that contains some information about
+the necessary headers. In our app, you'll need the `libcurl` library to make some HTTP calls. To create a
+definition file:
+
+1. Select the `src` folder and create a new directory with **File | New | Directory**.
+2. Name new directory **nativeInterop/cinterop**. This is the default convention for header file locations, though it can
+   be overridden in the `build.gradle` file if you use a different location.
+3. Select this new subfolder and create a new `libcurl.def` file with **File | New | File**.
+4. Update your file with the following code:
+
+```c
+headers = curl/curl.h
+headerFilter = curl/*
+
+compilerOpts.linux = -I/usr/include -I/usr/include/x86_64-linux-gnu
+linkerOpts.osx = -L/opt/local/lib -L/usr/local/opt/curl/lib -lcurl
+linkerOpts.linux = -L/usr/lib/x86_64-linux-gnu -lcurl
+```
+
+* The first entry is `headers`, the list of header files to generate Kotlin stubs for. You can add multiple files to
```

Comment:
This easily can be simplified to "`headers` contains the list of header files ...".
But the current version is also OK.